runtime/gateway.go

```diff
@@ -476,11 +473,11 @@ func (gateway *Gateway) setupMetrics(config *StaticConfig) (err error) {
 	gateway.AllHostScope = gateway.RootScope.SubScope(
```

Comment:
Is `AllHostScope` needed anymore?

Reply:
AllHostScope is needed since it is related to zap namespace, however, metrics would only have one scope which is RootScope

Reply:
> AllHostScope is needed since it is related to zap namespace and tchannel metrics namespace

Related in what way? Can you elaborate?

Reply:
https://github.com/uber/zanzibar/blob/master/runtime/gateway.go#L556-L560
https://github.com/uber/zanzibar/blob/master/runtime/gateway.go#L619-L622
https://github.com/uber/zanzibar/blob/master/runtime/gateway.go#L580-L596

> > AllHostScope is needed since it is related to zap namespace and tchannel metrics namespace
> 
> Related in what way? Can you elaborate?

https://github.com/uber/zanzibar/blob/master/runtime/gateway.go#L556-L560
https://github.com/uber/zanzibar/blob/master/runtime/gateway.go#L619-L622
https://github.com/uber/zanzibar/blob/master/runtime/gateway.go#L580-L596

Reply:
All these references can be replaced by the `RootScope`, the `AllHostScope` metric name `service + "." + env + ".all-workers"` is redundant and verbose, it is already tagged with `env` and `service`. I believe the original purpose for its existence is to reduce the cardinality introduced by tags `host` and `dc`, which I don't know if it is still a valid concern. If we want to have the same effect, we can replace it the `RootScope` before it is tagged with perHostTags, not the one that is overwritten here https://github.com/uber/zanzibar/pull/499/files#diff-ee4b13d64c39fd8e3d0483c1076bb31cR480.

Reply:
fixed